Source/Methods/Envelope Tracking/EnvelopeTracking.cs

```diff
@@ -0,0 +1,309 @@
+                {
+                    TerminalConfiguration = RfsgTerminalConfiguration.Differential,
+                };
+            }
+        }
+
+        public struct TrackerConfiguration
+        {
+            public double InputImpedance_Ohms;
+            public double CommonModeOffset_V;
+            public double Gain_VperV;
+            public double OutputOffset_V;
+
+            public static TrackerConfiguration GetDefault()
+            {
+                return new TrackerConfiguration()
+                {
+                    InputImpedance_Ohms = 1e6,
+                    CommonModeOffset_V = 1.0,
+                    Gain_VperV = 2.5,
+                    OutputOffset_V = 2.55
+                };
+            }
+        }
+
+        public enum DetroughType { 
+            Exponential, 
+            Cosine, 
+            Power
+        };
+
+        public struct DetroughConfiguration
+        {
+            public DetroughType Type;
+            public double MinimumVoltage;
+            public double MaximumVoltage;
+            public double Exponent;
+
+            public static DetroughConfiguration GetDefault()
+            {
+                return new DetroughConfiguration()
+                {
+                    Type = DetroughType.Exponential,
+                    MinimumVoltage = 1.5,
+                    MaximumVoltage = 3.5,
+                    Exponent = 1.2
+                };
+            }
+        }
+
+        public struct LookUpTable
+        {
+            public float[] DutInputPower_dBm;
+            public float[] SupplyVoltage_V;
+        }
+
+        public struct SynchronizationConfiguration
+        {
+            public double RFDelayRange_s;
+            public double RFDelay_s;
+
+            public static SynchronizationConfiguration GetDefault()
+            {
+                return new SynchronizationConfiguration()
+                {
+                    RFDelayRange_s = 2e-6, // +/- 1us
+                    RFDelay_s = 0.0
+                };
+            }
+        }
+        #endregion
+
+        #region Envelope Creation
+        public static Waveform CreateEnvelopeWaveform(Waveform referenceWaveform, DetroughConfiguration detroughConfig)
+        {
+            Waveform envelopeWaveform = new Waveform()
+            {
+                Name = referenceWaveform.Name + "Envelope",
+                Data = referenceWaveform.Data.Clone(),
+                SignalBandwidth_Hz = 0.8 * referenceWaveform.SampleRate,
+                PAPR_dB = double.NaN, // unnecessary to calculate as it won't be used
+                BurstLength_s = double.NaN, // not applicable in ET
+                SampleRate = referenceWaveform.SampleRate,
+                // burst start and stop locations will be left as null since they also will not be used
+                IdleDurationPresent = referenceWaveform.IdleDurationPresent, // can't null a bool, so copy from reference waveform

```

Comment:
Same as 101-102?

Reply:
hmm this on is interesting. i guess the envelope could have an idle point.
is your suggesting just not to touch it?

Reply:
3db6cc58ffac7217621bf71c67b55f8a0e08e876